server/index.js

```diff
@@ -30,6 +31,13 @@ export default class Server {
     this.config = getConfig(this.dir)
 
     this.defineRoutes()
+    babelRegister({
+      // If there's .babelrc file in the app path(or similar) it'll take the
+      // precedence over this.
+      presets: [require.resolve('./build/babel/preset')],
+      extensions: ['.js'],
+      only: this.config.transpileModules || []
+    })
```

Comment:
https://github.com/standard-things/esm looks promising. Maybe we could use this in production and `babel-register` in dev mode?